frontend/measure/src/app/details/details.component.ts

```diff
@@ -0,0 +1,172 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { ngxCsv } from 'ngx-csv/ngx-csv';
+import { ActivatedRoute } from '@angular/router';
+
+import { BackendService } from '../backend.service';
+import { Measurement } from '../classes/measurement';
+import { RhValue } from '../classes/rh-value';
+import { PaginatedList } from '../classes/paginated-list';
+
+import { Color } from 'ng2-charts';
+
+@Component({
+  selector: 'app-details',
+  templateUrl: './details.component.html',
+  styleUrls: ['./details.component.css'],
+})
+export class DetailsComponent implements OnInit {
+  measurement: Measurement;
+  rhValue: PaginatedList<RhValue>;
+  speed = '';
+  measurementData = [];
+  rhvalueData = [];
+  measurementName = '';
+  id = +this.route.snapshot.paramMap.get('id');
+
+  constructor(private route: ActivatedRoute, private backend: BackendService) {}
+
+  getMeasurement(): void {
+    this.backend.get_measurement(this.id).subscribe((measurement: Measurement) => {
+      if (measurement.steps_per_measurement === 20) {
+        this.speed = 'High Speed : Low Resolution';
+      } else if (measurement.steps_per_measurement === 10) {
+        this.speed = 'Medium Speed : Medium Resolution';
+      } else if (measurement.steps_per_measurement === 2) {
+        this.speed = 'Low Speed : High Resolution';
+      }
```

Comment:
Gjorde såhär nu, eftersom användaren ställer in speed i detta format.

Reply:
flytta arrayen (speeds? heter den kanske) till en gemensam plats (kanske i en ny fil i classes?) och använd den för att kolla upp värdet på strängarna. (Du kan tom skapa ett pipe för att göra formatteringen)